src/components/item/settings/GeolocationPicker.tsx

```diff
@@ -0,0 +1,153 @@
+import { ChangeEventHandler, useCallback, useEffect, useState } from 'react';
+
+import Clear from '@mui/icons-material/Clear';
+import {
+  Box,
+  IconButton,
+  LinearProgress,
+  List,
+  ListItemButton,
+  Stack,
+  TextField,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+
+import { DiscriminatedItem } from '@graasp/sdk';
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { OpenStreetMapProvider } from 'leaflet-geosearch';
+
+import { useBuilderTranslation } from '@/config/i18n';
+import { hooks, mutations } from '@/config/queryClient';
+import { BUILDER } from '@/langs/constants';
+import useDebouncedCallback from '@/utils/useDebounce';
+
+type SearchResult = {
+  osm_id: string;
+  raw: { lat: number; lon: number };
+  label: string;
+};
+
+const provider = new OpenStreetMapProvider();
+
+const GeolocationPicker = ({
+  item,
+}: {
+  item: DiscriminatedItem;
+}): JSX.Element => {
+  const { t } = useBuilderTranslation();
+  const { data: geoloc } = hooks.useItemGeolocation(item.id);
+  const [query, setQuery] = useState<string | null>(null);
+  const { mutate: putGeoloc } = mutations.usePutItemGeolocation();
+  const { mutate: deleteGeoloc } = mutations.useDeleteItemGeolocation();
+  const [loading, setLoading] = useState(false);
+  const [results, setResults] = useState<SearchResult[]>([]);
+
+  useEffect(() => {
+    if (geoloc) {
+      setQuery(geoloc.address);
+    }
+  }, [geoloc]);
+
+  const callback = useCallback(() => {
+    if (query && query !== geoloc?.address) {
+      setLoading(true);
+      provider
+        .search({ query })
+        .then((e) => {
+          setResults(e as unknown as SearchResult[]);
+        })
+        .finally(() => {
+          setLoading(false);
+        });
+    }
+  }, [query, geoloc]);
+
+  const { isDebounced } = useDebouncedCallback(callback, 1000);
+
+  const onChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    setQuery(e.target.value);
+  };
+
+  const onChangeOption = (option: SearchResult): void => {
+    const {
+      raw: { lat, lon: lng },
+      label,
+    } = option;
+    putGeoloc({
+      itemId: item.id,
+      geolocation: { lng, lat, address: label },
+    });
+    setResults([]);
+  };
+
+  const clearGeoloc = () => {
+    deleteGeoloc({ itemId: item.id });
+  };
+
+  const isDisabled = Boolean(geoloc && geoloc?.item?.id !== item.id);
```

Comment:
```suggestion
  const isDisabled = geoloc?.item?.id !== item.id;
```

Reply:
I want it to be enabled if geoloc is undefined. I'll add a comment ;)